CI: playwright-shiny shards fail on syrupy "unused snapshots" even when all tests pass

> [!NOTE]
> This issue was researched and written by Claude Code after observing the failure on an unrelated docs-only PR (#2332).

## Summary

`playwright-shiny` CI shards are failing with exit code 1 from syrupy's unused-snapshot check even though **every test in the shard passes**. The failure occurs on `main` (scheduled and push runs) as well as on PRs, and it did **not** recover when the failed jobs were restarted, so this behaves as persistent breakage rather than a pass-on-rerun flake.

## Failure signature

The job log ends like this (all green, then a nonzero exit):

```
--------------------------- snapshot report summary ----------------------------
2 snapshots passed. 10 snapshots unused.
Re-run pytest with --snapshot-update to delete unused snapshots.
==================== 74 passed, 1 rerun in 94.45s (0:01:34) ====================
make[1]: *** [Makefile:234: playwright] Error 1
make: *** [Makefile:244: playwright-shiny] Error 2
##[error]Process completed with exit code 2.
```

Other affected shards show `12 snapshots unused.` with `90 passed` or `79 passed`.

## Affected runs

- PR #2332 (docs-only change), run [28878032396](https://github.com/posit-dev/py-shiny/actions/runs/28878032396): `playwright-shiny (3.14, chromium)` shards 0, 1, and 2 failed; shard 3 passed. Restarting the failed jobs reproduced the identical failure ([shard 0 rerun job](https://github.com/posit-dev/py-shiny/actions/runs/28878032396/job/85659790185)).
- `main` scheduled run [28854865414](https://github.com/posit-dev/py-shiny/actions/runs/28854865414) (2026-07-07): same signature on `playwright-shiny (3.10, chromium, 1)` (`2 snapshots passed. 10 snapshots unused.`, `79 passed`) and `playwright-shiny (3.13, firefox, 2)` (`12 snapshots unused.`, `70 passed, 3 skipped`).
- `main` push run [28814680090](https://github.com/posit-dev/py-shiny/actions/runs/28814680090) (merge of #2331) also failed, though that run additionally shows a real single-test failure on webkit.

## Analysis

The snapshot counts point at `tests/playwright/shiny/components/data_frame/example/__snapshots__/test_data_frame.ambr`, which contains exactly **12 snapshots** (matching the `12 unused` / `2 passed + 10 unused` arithmetic). The snapshot assertions live in parametrized tests in `tests/playwright/shiny/components/data_frame/example/test_data_frame.py` (pandas/polars parametrization).

The `playwright-shiny` workflow shards tests with `--num-shards 4 --shard-id N` (`.github/workflows/pytest.yaml`), which can distribute the parametrized test items of a single module across different shards. A shard that runs only some (or none) of the module's snapshot-asserting items sees the remaining `.ambr` entries as "unused", and syrupy fails the session with exit code 1 even though every executed test passed.

## Possible directions

- Make syrupy's unused-snapshot check non-fatal (warning) for sharded CI runs, keeping it strict for unsharded/local runs.
- Keep all test items of snapshot-using modules within a single shard.
- Move/refactor the snapshot assertions so each shard's collected set is self-consistent.

## Staleness

Per `.claude/references/commit-conventions.md`: close if no new occurrences are reported for 30 days (though note this one currently reproduces deterministically, so it likely needs a fix rather than aging out).

--- a/Makefile
+++ b/Makefile
@@ -230,8 +230,13 @@ install-rsconnect: FORCE
 # All end-to-end tests with playwright
 # `--timeout` bounds each test so a hang fails fast with a thread-stack dump
 # instead of consuming the whole job.
+# `--snapshot-warn-unused` because CI runs each browser and pytest-shard shard
+# as a separate partial job; syrupy (>=5.5.0, under xdist) would otherwise fail
+# the job for snapshots that belong to the tests running in the *other* jobs.
+# Stale snapshots can still be cleaned up with an all-browser, unsharded run
+# using `--snapshot-update`.
 playwright: install-playwright ## All end-to-end tests with playwright; (TEST_FILE="" from root of repo)
-	pytest $(PLAYWRIGHT_VERBOSE) --timeout=$(PLAYWRIGHT_TEST_TIMEOUT) --timeout-method=$(PLAYWRIGHT_TEST_TIMEOUT_METHOD) -o faulthandler_timeout=$(PLAYWRIGHT_FAULTHANDLER_TIMEOUT) $(TEST_FILE) $(PYTEST_BROWSERS)
+	pytest $(PLAYWRIGHT_VERBOSE) --snapshot-warn-unused --timeout=$(PLAYWRIGHT_TEST_TIMEOUT) --timeout-method=$(PLAYWRIGHT_TEST_TIMEOUT_METHOD) -o faulthandler_timeout=$(PLAYWRIGHT_FAULTHANDLER_TIMEOUT) $(TEST_FILE) $(PYTEST_BROWSERS)
 
 playwright-debug: install-playwright ## All end-to-end tests, chrome only, headed; (TEST_FILE="" from root of repo)
 	pytest -c tests/playwright/playwright-pytest.ini $(TEST_FILE)

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -86,7 +86,7 @@ test = [
     "pytest-rerunfailures",
     "pytest-cov",
     "coverage",
-    "syrupy>=4.7.1",
+    "syrupy>=5.5.1",
     "psutil",
     "astropy",
     "suntime",

